### ISSUE TITLE
IIS Applications hosting (in site-folder) is not working for Blazor Server deployment

### ISSUE THREAD

**Issue by LarsOutzen** [dotnet/aspnetcore#43161]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

See https://docs.microsoft.com/en-us/aspnet/core/blazor/host-and-deploy/?view=aspnetcore-6.0&tabs=visual-studio#app-base-path
Try to test "Option 1" OR "Option 2" on IIS for your Blazor Server template projects' Counter

cc: @guardrex https://github.com/dotnet/AspNetCore.Docs/issues/26667

### Expected Behavior

A working site/CoolApp/

### .NET Version

Azure VM: dotnet --info  Host (useful for support):   Version: 6.0.3   Commit:  c24d9a9c91

### Anything else?

On my dev machine I have Microsoft.AspNetCore.App 6.0.7

**Comment by LarsOutzen** [comment]
No - it is a plain DataCenter 2016 VM with IIS (and SQL Express) and no other Azure services ..

**Comment by guardrex** [comment]
Are WebSockets enabled? ...

https://docs.microsoft.com/en-us/aspnet/core/fundamentals/websockets?view=aspnetcore-6.0#enabling-websockets-on-iis

**Comment by LarsOutzen** [comment]
Yes - ![image](https://user-images.githubusercontent.com/7156357/183652854-25b2a4bd-f048-4c31-bca6-b5851e47bc9b.png)
We are actually very happy with Blazor https://www.youtube.com/watch?v=2OsWFpWUhJk&t=35s even have an hybrid MAUI app https://play.google.com/store/apps/details?id=com.companyname.scsome.mobileapp and iOS on the way today

**Comment by TanayParikh** [comment]
Hi @LarsOutzen, thanks for reaching out. Can you please clarify what exact issue you're running into using Blazor Server and IIS?

**Comment by LarsOutzen** [comment]
I am looking for at way to host at CoolApp on an existing (Umbraco CMS) site/CoolApp/
According to https://docs.microsoft.com/en-us/aspnet/core/blazor/host-and-deploy/?view=aspnetcore-6.0&tabs=visual-studio#app-base-path there are "Option 1" OR "Option 2" - but none of them work

**Comment by SteveSandersonMS** [comment]
@LarsOutzen Thanks. Unfortunately we still don't know what specific problem you encounter. Is there an error message?

One way we might be able to investigate is if you give us exact repro steps, i.e., how to recreate the same problem starting from the default project template. This includes specifying each step in configuring IIS. Would you be able to provide that info?

**Comment by LarsOutzen** [comment]
New default Windows (Windows Server 2022 Datacenter Azure Edition) 
    + IIS defaults except for WebSockets that had to be added manually
![image](https://user-images.githubusercontent.com/7156357/184351109-93059d4f-e3f7-4777-a512-04af891f8014.png)
![image](https://user-images.githubusercontent.com/7156357/184351223-ccb8e96d-17da-451b-8406-6c6a0b1e163e.png)

![image](https://user-images.githubusercontent.com/7156357/184352858-4e88f882-9c73-4b61-92d0-86ba2929332b.png)

```
dotnet --info

global.json file:
  Not found

Host:
  Version:      6.0.8
  Architecture: x64
  Commit:       55fb7ef977

.NET SDKs installed:
  No SDKs were found.

.NET runtimes installed:
  Microsoft.AspNetCore.App 3.1.28 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.28 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.8 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]

Download .NET:
  https://aka.ms/dotnet-download

Learn about .NET Runtimes and SDKs:
  https://aka.ms/dotnet/runtimes-sdk-info
```

End of modified Program.cs
```
app.UseHttpsRedirection();

app.UseStaticFiles("/CoolApp");

app.UsePathBase("/CoolApp");
// ...
app.UseRouting();

app.MapBlazorHub();
app.MapFallbackToPage("/_Host");

app.Run();
```

**Comment by guardrex** [comment]
@LarsOutzen ... That looks like a failure to even start the app in IIS, not an app base path problem.

We're vaguely&dagger; guiding devs to use the *Publish to IIS* guidance for Blazor Server apps ...

https://docs.microsoft.com/aspnet/core/tutorials/publish-to-iis

... which includes the prohibition on virtual directories and doesn't guide devs to convert to an ASP.NET Application.

&dagger;vaguely ... Several spots across Blazor docs refer devs to the main doc set for Blazor Server; but in this particular case, I left the docs issue open because I detect that some improvements can be made on where and how we cross-link IIS coverage.

I'm 👂 for further PU remarks here to include on the docs issue.

**Comment by guardrex** [comment]
@SteveSandersonMS @TanayParikh ... I do have a ❓ on middleware placement.

We made an adjustment a while back to tell devs ...

> Call `UsePathBase` **_early_** in the app's request processing pipeline ...

... and show calling it prior to `UseRouting`. **_However_**, can't we be more explicit on **_how early_** to call `UsePathBase`? For example, can I tell devs to 'call it first' (i.e., immediately after the `WebApplicationBuilder` is built)?

**Comment by LarsOutzen** [comment]
.. the .NET Core Hosting Bundle was already installed on that server-type, however it did not work until it was "Repaired"
The root WebSite works fine with Blazor, however the sub-app is still broken:
![image](https://user-images.githubusercontent.com/7156357/184360834-0271b95b-82c4-48ed-adee-884e440648b5.png)

**Comment by guardrex** [comment]
I see. I **_did just notice_** that we guide devs for **Convert to Application** for sub-apps. It's been a while since I've looked at this scenario ... **_several years!_**

> Right-click the sub-app folder in IIS Manager and select **Convert to Application**.

https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/iis/advanced?view=aspnetcore-6.0#sub-applications

**Comment by javiercn** [comment]
Some of the write-up [here](https://github.com/dotnet/aspnetcore/issues/43191#issuecomment-1212156106) might also help.

@guardrex feel free to make an honest doc out of my comment 😄

**Comment by guardrex** [comment]
We started with basic app base path guidance. In a series of perhaps four reader doc issues over the last couple of years, we improved the guidance with each reader leaving happy in the end with a working app 🎉.

I see from your post, @javiercn, that we need *another round of updates*. @LarsOutzen, I'm going to use your issue to work on it, along with the other items that I've identified. However, there will be a delay. I'm working .NET 7 doc issues, which have a high priority. When RC1 lands, I'm going to be very busy 🏃, so it's likely that doc work on this will be more EOY or 23Q1. We'll see.

@javiercn ... One specific ❓ that I don't see you mention: For those scenarios where the current guidance works, can I tell devs when they place `UsePathBase` to place it immediately after the `WebApplicationBuilder` is built, even before HTTPS Redirection Middleware? Merely saying "call it early" isn't specific and doesn't seem particularly helpful.
